auth.go

```diff
@@ -426,77 +456,120 @@ func (x *Central) Logout(sessionkey string) error {
+	if err := x.userStore.SetPassword(userId, password); err != nil {
+		x.Log.Infof("SetPassword failed (%v) (%v)", userId, password)
 		return err
 	}
-	x.Log.Infof("SetPassword successful (%v)", identity)
-	return x.sessionDB.InvalidateSessionsForIdentity(identity)
+	x.Log.Infof("SetPassword successful (%v)", userId)
+	return x.sessionDB.InvalidateSessionsForIdentity(userId)
 }
 
 // Create a one-time token that can be used to reset the password with a subsequent call to ResetPasswordFinish.
 // Any subsequent call to ResetPasswordStart causes the current token to be invalidated, so there can only
 // be a single active token. The token is valid until the time specified by 'expires'.
-func (x *Central) ResetPasswordStart(identity string, expires time.Time) (string, error) {
-	token, err := x.authenticator.ResetPasswordStart(identity, expires)
+func (x *Central) ResetPasswordStart(userId UserId, expires time.Time) (string, error) {
+	token, err := x.userStore.ResetPasswordStart(userId, expires)
 	if err != nil {
-		x.Log.Infof("ResetPasswordStart failed (%v) (%v)", identity, err)
+		x.Log.Infof("ResetPasswordStart failed (%v) (%v)", userId, err)
 		return "", err
 	}
-	x.Log.Infof("ResetPasswordStart successful (%v)", identity)
+	x.Log.Infof("ResetPasswordStart successful (%v)", userId)
 	return token, err
 }
 
 // Complete the password reset process, by providing a token that was generated by ResetPasswordStart.
 // If this succeeds, then the password is set to 'password', and the token becomes invalid.
-func (x *Central) ResetPasswordFinish(identity string, token string, password string) error {
-	if err := x.authenticator.ResetPasswordFinish(identity, token, password); err != nil {
-		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", identity, err)
+func (x *Central) ResetPasswordFinish(userId UserId, token string, password string) error {
+	if err := x.userStore.ResetPasswordFinish(userId, token, password); err != nil {
+		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", userId, err)
 		return err
 	}
-	x.Log.Infof("ResetPasswordFinish successful (%v)", identity)
-	x.sessionDB.InvalidateSessionsForIdentity(identity)
+	x.Log.Infof("ResetPasswordFinish successful (%v)", userId)
+	x.sessionDB.InvalidateSessionsForIdentity(userId)
 	return nil
 }
 
-// Create an identity in the Authenticator.
-// For the equivalent operation in the PermitDB, simply call SetPermit()
-func (x *Central) CreateAuthenticatorIdentity(identity, password string) error {
-	e := x.authenticator.CreateIdentity(identity, password)
+// Create an identity in the AuthUserStore.
+func (x *Central) CreateUserStoreIdentity(email, username, firstname, lastname, mobilenumber, password string) error {
+	e := x.userStore.CreateIdentity(email, username, firstname, lastname, mobilenumber, password)
+	if e == nil {
+		x.Log.Infof("CreateAuthenticatorIdentity successful (%v)", email)
+	} else {
+		x.Log.Infof("CreateAuthenticatorIdentity failed (%v) (%v)", email, e)
```

Comment:
Failure here should be an 'Error', and not an 'Info' message, because new users will always be created by an 'admin' user, so we expect this to succeed.
Also, it would probably be good to include the UserId in the success message. From here on out, most of the log messages will refer to the UserId and not the email, so this is the only point where one can sync the two up, if all you have is the logs. Perhaps `CreateIdentity` should return `(UserId, error)`?